Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -1432,6 +1432,17 @@ void Document::set_visited_link_color(Color color)
     m_visited_link_color = color;
 }
 
+Optional<Vector<String> const&> Document::supported_color_schemes() const
+{
+    return m_supported_color_schemes;
+}
+
+void Document::set_supported_color_schemes(Vector<String> color_schemes)
+{
+    if (!m_supported_color_schemes.has_value())
+        m_supported_color_schemes = color_schemes;
```

Comment:
I think I get why this only sets once (because the algorithm in `HTMLMetaElement::inserted()` wants to only use the first matching `<meta>`'s value) but it's not quite correct: If the chosen `<meta name="color-scheme">` is removed, modified, or is no longer the first one, then we continue to use its value instead of the new first one (or nothing).

IMO, this setter should always set, and that code in `HTMLMetaElement::inserted()` should be moved into a `Document::obtain_supported_color_schemes()` method or similar, which `HTMLMetaElement::inserted()` just calls. I think this is also true for the `theme-color` code that was already there.

Reply:
Yeah, a complete implementation would also need to recalculate the supported color schemes on removal and on attribute changes.

Reply:
Done, although it doesn't actually have much useful observable behaviour since the styles aren't invalidated correctly (or at all).